Add the capability to Freescale MX2 framebuffer to power on/off LCD's
backlight.

Signed-off-by: Julien Boibessot <julien.boibessot@armadeus.com>

Index: linux-2.6.29.6/arch/arm/plat-mxc/include/mach/imx_fb.h
===================================================================
--- linux-2.6.29.6.orig/arch/arm/plat-mxc/include/mach/imx_fb.h	2010-01-27 18:50:48.000000000 +0100
+++ linux-2.6.29.6/arch/arm/plat-mxc/include/mach/imx_fb.h	2010-01-27 18:51:43.000000000 +0100
@@ -28,6 +28,7 @@
 	int (*init)(struct platform_device *pdev);
 	int (*exit)(struct platform_device *pdev);
 
+	void (*power_backlight)(struct platform_device *pdev, int on);
 	int (*power_lcd)(struct platform_device *pdev, int on);
 };
 
Index: linux-2.6.29.6/drivers/video/mxc/mx2fb.c
===================================================================
--- linux-2.6.29.6.orig/drivers/video/mxc/mx2fb.c	2010-01-27 18:54:43.000000000 +0100
+++ linux-2.6.29.6/drivers/video/mxc/mx2fb.c	2010-01-27 18:56:36.000000000 +0100
@@ -772,6 +772,9 @@
 		mx2fbi->pdata->init(mx2fbi->pdev);
 		if (mx2fbi->pdata->power_lcd)
 			mx2fbi->pdata->power_lcd(mx2fbi->pdev, 1);
+		if (mx2fbi->pdata->power_backlight)
+			mx2fbi->pdata->power_backlight(mx2fbi->pdev, 1);
+
 		_set_brightness(brightness);
 		fb_enabled++;
 #ifdef CONFIG_FB_MXC_TVOUT
@@ -800,6 +803,8 @@
 	else {
 		if (fb_enabled) {
 			mx2fbi->pdata->exit(mx2fbi->pdev);
+			if (mx2fbi->pdata->power_backlight)
+				mx2fbi->pdata->power_backlight(mx2fbi->pdev, 0);
 			if (mx2fbi->pdata->power_lcd)
 				mx2fbi->pdata->power_lcd(mx2fbi->pdev, 0);
 			_set_brightness(0);
